script.js

```diff
@@ -15,4 +15,28 @@ function generateGrid(pixles)
         item.addEventListener('mouseover', () => { item.style.backgroundColor = 'black'; });
     })
 }
-generateGrid(16);
\ No newline at end of file
+generateGrid(16);
+//reset button
+function clearGrid() {
+    document.querySelectorAll('.grid-item').forEach(item => item.style = '')
+  };
+  document.querySelector(".reset-btn").addEventListener('click', clearGrid);
+
+// New grid button
+document.querySelector(".grid-btn").addEventListener("click",newGrid)
+function newGrid() {
+       
+        let differentGridSize = prompt("Enter a new grid size(1-100):");
+        if (differentGridSize >= 1 && differentGridSize <= 100) {
+            generateGrid(differentGridSize);
+            clearGrid(differentGridSize)
+        } else {
+        alert("please enter a size  within 1-100 pixles")  
+        }
```

Comment:
Have just noticed that ! 👀  
I wrote a longer if statements which calls the function clearGrid ( )  when (else ) happens 
Also you made me pay attention to adding another if case when the user enters a letter instead of a number.